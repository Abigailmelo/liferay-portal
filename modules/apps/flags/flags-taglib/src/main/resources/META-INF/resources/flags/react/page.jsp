<%--
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
--%>

<%@ include file="/flags/react/init.jsp" %>

<%
String companyName = (String)request.getAttribute("liferay-flags:flags:companyName");
JSONObject dataJSONObject = (JSONObject)request.getAttribute("liferay-flags:flags:dataJSONObject");
String elementClasses = (String)request.getAttribute("liferay-flags:flags:elementClasses");
boolean enabled = GetterUtil.getBoolean(request.getAttribute("liferay-flags:flags:enabled"), true);
boolean flagsEnabled = GetterUtil.getBoolean(request.getAttribute("liferay-flags:flags:flagsEnabled"), true);
String id = StringUtil.randomId() + StringPool.UNDERLINE + "id";
String message = (String)request.getAttribute("liferay-flags:flags:message");
boolean onlyIcon = GetterUtil.getBoolean(request.getAttribute("liferay-flags:flags:onlyIcon"));
Map<String, String> reasons = (Map<String, String>)request.getAttribute("liferay-flags:flags:reasons");
boolean signedIn = (boolean)request.getAttribute("liferay-flags:flags:signedIn");
String uri = (String)request.getAttribute("liferay-flags:flags:uri");

if (Validator.isNull(message)) {
	message = LanguageUtil.get(resourceBundle, "report");
}
%>

<div class="taglib-flags <%= Validator.isNotNull(elementClasses) ? elementClasses : "" %>" id="<%= id %>">
	<c:choose>
		<c:when test="<%= onlyIcon %>">
			<clay:button
				disabled="<%= true %>"
				elementClasses="btn-outline-borderless btn-outline-secondary lfr-portal-tooltip"
				icon="flag-empty"
				monospaced="<%= true %>"
				size="sm"
				style="secondary"
				title="<%= message %>"
			/>
		</c:when>
		<c:otherwise>
			<clay:button
				disabled="<%= true %>"
				elementClasses="btn-outline-borderless btn-outline-secondary"
				icon="flag-empty"
				label="<%= message %>"
				size="sm"
				style="secondary"
			/>
		</c:otherwise>
	</c:choose>

	<%
	Map<String, Object> context = new HashMap<>();
	context.put("namespace", PortalUtil.getPortletNamespace(PortletKeys.FLAGS));

	Map<String, Object> props = new HashMap<>();
	props.put("baseData", dataJSONObject);
	props.put("companyName", companyName);
	props.put("disabled", !enabled);
	props.put("forceLogin", !flagsEnabled);

	if (Validator.isNotNull(message)) {
		props.put("message", message);
	}

	props.put("onlyIcon", onlyIcon);
	props.put("pathTermsOfUse", PortalUtil.getPathMain() + "/portal/terms_of_use");
	props.put("reasons", reasons);
	props.put("signedIn", signedIn);
	props.put("uri", uri);

	Map<String, Object> data = new HashMap<>();
	data.put("props", props);
	data.put("context", context);
	%>

	<react:component
		data="<%= data %>"
		module="flags/react/js/index.es"
	/>
</div>