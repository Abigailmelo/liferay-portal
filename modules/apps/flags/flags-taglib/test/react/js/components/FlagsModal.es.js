/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import FlagsModal from '../../../../src/main/resources/META-INF/resources/flags/react/js/components/FlagsModal.es';
import React from 'react';
import ReactDOM from 'react-dom';
import {
	STATUS_ERROR,
	STATUS_LOGIN,
	STATUS_REPORT,
	STATUS_SUCCESS
} from '../../../../src/main/resources/META-INF/resources/flags/react/js/constants.es';

beforeAll(() => {
	ReactDOM.createPortal = jest.fn(element => {
		return element;
	});
});

afterEach(() => {
	jest.restoreAllMocks();
});

function _renderFlagsModalComponent({
	companyName = 'Liferay',
	handleClose = () => {},
	handleInputChange = () => {},
	handleSubmit = () => {},
	isSending = false,
	pathTermsOfUse = '/',
	selectedReason = 'value',
	reasons = {value: 'text', value2: 'text2'},
	signedIn = true,
	status = STATUS_REPORT
} = {}) {
	return render(
		<FlagsModal
			companyName={companyName}
			handleClose={handleClose}
			handleInputChange={handleInputChange}
			handleSubmit={handleSubmit}
			isSending={isSending}
			pathTermsOfUse={pathTermsOfUse}
			reasons={reasons}
			selectedReason={selectedReason}
			signedIn={signedIn}
			status={status}
		/>
	);
}

describe('FlagsModal', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {asFragment} = _renderFlagsModalComponent();

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders as guess and render email field', () => {
		const {getByLabelText} = _renderFlagsModalComponent({
			signedIn: false
		});

		expect(getByLabelText('email', {exact: false})).not.toBe(null);
	});

	it('renders error', () => {
		const {asFragment} = _renderFlagsModalComponent({status: STATUS_ERROR});

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders login', () => {
		const {asFragment} = _renderFlagsModalComponent({status: STATUS_LOGIN});

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders success', () => {
		const {asFragment} = _renderFlagsModalComponent({
			status: STATUS_SUCCESS
		});

		expect(asFragment()).toMatchSnapshot();
	});
});
