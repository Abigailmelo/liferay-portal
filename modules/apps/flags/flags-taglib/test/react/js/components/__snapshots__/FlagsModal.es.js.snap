// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagsModal renders 1`] = `
<DocumentFragment>
  <div
    class="modal-backdrop fade"
  />
  <div
    class="fade modal d-block"
  >
    <div
      class="modal-dialog modal-md"
      tabindex="-1"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            report-inappropriate-content
          </div>
          <button
            aria-label="close"
            class="close btn btn-unstyled"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-times"
              role="presentation"
            >
              <use
                xlink:href="#times"
              />
            </svg>
          </button>
        </div>
        <form>
          <div
            class="modal-body"
          >
            <p>
              you-are-about-to-report-a-violation-of-our-x.-all-reports-are-strictly-confidential
            </p>
            <div
              class="form-group"
            >
              <label
                class="control-label"
                for="selectedReason"
              >
                reason-for-the-report
              </label>
              <select
                class="form-control"
                id="selectedReason"
                name="selectedReason"
              >
                <option
                  value="value"
                >
                  text
                </option>
                <option
                  value="value2"
                >
                  text2
                </option>
                <option
                  value="other"
                >
                  other-reason
                </option>
              </select>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <div
              class="modal-item-first"
            />
            <div
              class="modal-item"
            />
            <div
              class="modal-item-last"
            >
              <div
                class="btn-group"
                role="group"
              >
                <div
                  class="btn-group-item"
                >
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    cancel
                  </button>
                </div>
                <div
                  class="btn-group-item"
                >
                  <button
                    class="btn btn-primary"
                    type="submit"
                  >
                    report
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FlagsModal renders error 1`] = `
<DocumentFragment>
  <div
    class="modal-backdrop fade"
  />
  <div
    class="fade modal d-block"
  >
    <div
      class="modal-dialog modal-md"
      tabindex="-1"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            report-inappropriate-content
          </div>
          <button
            aria-label="close"
            class="close btn btn-unstyled"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-times"
              role="presentation"
            >
              <use
                xlink:href="#times"
              />
            </svg>
          </button>
        </div>
        <div
          class="modal-body"
        >
          <p>
            <strong>
              an-error-occurred-while-sending-the-report.-please-try-again-in-a-few-minutes
            </strong>
          </p>
        </div>
        <div
          class="modal-footer"
        >
          <div
            class="modal-item-first"
          />
          <div
            class="modal-item"
          />
          <div
            class="modal-item-last"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FlagsModal renders login 1`] = `
<DocumentFragment>
  <div
    class="modal-backdrop fade"
  />
  <div
    class="fade modal d-block"
  >
    <div
      class="modal-dialog modal-md"
      tabindex="-1"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            report-inappropriate-content
          </div>
          <button
            aria-label="close"
            class="close btn btn-unstyled"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-times"
              role="presentation"
            >
              <use
                xlink:href="#times"
              />
            </svg>
          </button>
        </div>
        <div
          class="modal-body"
        >
          <p>
            <strong>
              please-sign-in-to-flag-this-as-inappropriate
            </strong>
          </p>
        </div>
        <div
          class="modal-footer"
        >
          <div
            class="modal-item-first"
          />
          <div
            class="modal-item"
          />
          <div
            class="modal-item-last"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FlagsModal renders success 1`] = `
<DocumentFragment>
  <div
    class="modal-backdrop fade"
  />
  <div
    class="fade modal d-block"
  >
    <div
      class="modal-dialog modal-md"
      tabindex="-1"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            report-inappropriate-content
          </div>
          <button
            aria-label="close"
            class="close btn btn-unstyled"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-times"
              role="presentation"
            >
              <use
                xlink:href="#times"
              />
            </svg>
          </button>
        </div>
        <div
          class="modal-body"
        >
          <p>
            <strong>
              thank-you-for-your-report
            </strong>
          </p>
          <p>
            although-we-cannot-disclose-our-final-decision,-we-do-review-every-report-and-appreciate-your-effort-to-make-sure-x-is-a-safe-environment-for-everyone
          </p>
        </div>
        <div
          class="modal-footer"
        >
          <div
            class="modal-item-first"
          />
          <div
            class="modal-item"
          />
          <div
            class="modal-item-last"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
